src/convert_image/sse2.rs

```diff
@@ -576,6 +586,165 @@ fn yuv_to_lrgb(
     true
 }
 
+#[inline(always)]
+fn i420_to_lrgb(
+    width: u32,
+    height: u32,
+    last_src_plane: usize,
+    src_strides: &[usize],
+    src_buffers: &[&[u8]],
+    _last_dst_plane: usize,
+    dst_strides: &[usize],
+    dst_buffers: &mut [&mut [u8]],
+    channels: PixelFormatChannels,
+    colorimetry: Colorimetry,
+) -> bool {
+    if last_src_plane != 2
+        || last_src_plane >= src_strides.len()
+        || last_src_plane >= src_buffers.len()
+        || dst_strides.is_empty()
+        || dst_buffers.is_empty()
+    {
+        return false;
+    }
+
+    let depth = channels as usize;
+    let col_count = width as usize;
+    let line_count = height as usize;
+    let packed_rgb_stride = depth * col_count;
+
+    let y_stride = if src_strides[0] != 0 {
+        src_strides[0]
+    } else {
+        col_count
+    };
+
+    let u_stride = if src_strides[1] != 0 {
+        src_strides[1]
+    } else {
+        col_count / 2
+    };
+
+    let v_stride = if src_strides[2] != 0 {
+        src_strides[2]
+    } else {
+        col_count / 2
+    };
+
+    let rgb_stride = if dst_strides[0] == 0 {
+        packed_rgb_stride
+    } else {
+        dst_strides[0]
+    };
+
+    let rgb_plane = &mut dst_buffers[0];
+    let (y_plane, u_plane, v_plane) = (src_buffers[0], src_buffers[1], src_buffers[2]);
+
+    if line_count == 0 {
+        return true;
+    }
+
+    let max_stride = usize::max_value() / line_count;
+    if (y_stride > max_stride)
+        || (u_stride > max_stride)
+        || (v_stride > max_stride)
+        || (rgb_stride > max_stride)
+    {
+        return false;
+    }
+
+    let wg_height = line_count / 2;
+    if y_stride * line_count > y_plane.len()
+        || u_stride * wg_height > u_plane.len()
+        || v_stride * wg_height > v_plane.len()
+        || rgb_stride * line_count > rgb_plane.len()
+    {
+        return false;
+    }
+
+    let col = colorimetry as usize;
+    unsafe {
+        let xxym = _mm_set1_epi16(BACKWARD_WEIGHTS[col][0]);
+        let rcrm = _mm_set1_epi16(BACKWARD_WEIGHTS[col][1]);
+        let gcrm = _mm_set1_epi16(BACKWARD_WEIGHTS[col][2]);
+        let gcbm = _mm_set1_epi16(BACKWARD_WEIGHTS[col][3]);
+        let bcbm = _mm_set1_epi16(BACKWARD_WEIGHTS[col][4]);
+        let rn = _mm_set1_epi16(BACKWARD_WEIGHTS[col][5]);
+        let gp = _mm_set1_epi16(BACKWARD_WEIGHTS[col][6]);
+        let bn = _mm_set1_epi16(BACKWARD_WEIGHTS[col][7]);
+
+        let y_group = y_plane.as_ptr();
+        let u_group = u_plane.as_ptr();
+        let v_group = v_plane.as_ptr();
+        let rgb_group = rgb_plane.as_mut_ptr();
+        let rgb_depth = 2 * YUV_TO_LRGB_WAVES;
+        let nv12_depth = YUV_TO_LRGB_WAVES;
```

Comment:
Same as avx2. Check if `i420_depth` or `chroma_depth` is a better name for this.